source/imap/auth.d

```diff
@@ -16,20 +16,24 @@ string authCramMD5(string user, string pass, string challenge) {
     import std.array : array;
     import std.string : join;
     import std.format : format;
+    import std.exception : enforce;
     size_t n;
     uint i;
     ubyte[] resp, ret;
     ubyte[EVP_MAX_MD_SIZE] md;
     uint mdlen;
-    HMAC_CTX hmac;
 
     n = challenge.length * 3 / 4 + 1;
     resp.length = n;
     EVP_DecodeBlock(resp.ptr, cast(const(ubyte) *) challenge.toStringz, challenge.length.to!int);
 
-    HMAC_Init(&hmac, cast(const(void) *) pass.toStringz, pass.length.to!int, EVP_md5());
-    HMAC_Update(&hmac, cast(const(ubyte) *) resp.ptr, resp.length.to!int);
-    HMAC_Final(&hmac, md.ptr, &mdlen);
+    HMAC_CTX* hmac = HMAC_CTX_new();
```

Comment:
Do we also want to enforce a minimum `OpenSSL` version ?

Reply:
`HMAC_CTX_new()` isn't guarded, so I assumed it would be available for all versions - is that not the case?